File: 1002-maximum-width-ramp/maximum-width-ramp.cpp
class Solution {
public:
    int maxWidthRamp(vector<int>& nums) {
        vector<int>arr;
        int ans =0;
        int n = nums.size();
        arr.push_back(0);                                
        for(int i=1;i<n;i++){
            if(nums[i]<nums[arr.back()]){
                arr.push_back(i);
            }
        }

        for(int j=n-1;j>=0;j--){
            while(arr.size()>0 && nums[arr.back()]<=nums[j]){
                ans=max(ans,j-arr.back());
                arr.pop_back();
            }
        }
        return ans;
    }
};

File: 1002-maximum-width-ramp/maximum-width-ramp.go
func maxWidthRamp(nums []int) int {
    arr:=[]int{}
    ans:= 0
    for i,num:= range nums{
        if len(arr)==0 || nums[arr[len(arr)-1]]>num{
            arr=append(arr,i)
        }
    }   

    for j:=len(nums)-1; j>=0; j--{
        for len(arr)>0 && nums[arr[len(arr)-1]]<=nums[j]{
            if ans<j-arr[len(arr)-1]{
                ans=j-arr[len(arr)-1]
            }
            arr=arr[:len(arr)-1]
        }
    }
    return ans
}